File: challenges/3-tipos-de-triangulo/java/maikella/TiposDeTriangulo.java
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;
import static java.lang.Math.pow;

public class TiposDeTriangulo {
   public static void main(String[] args) {
      start();
   }

   private static void start (){

      Scanner s = new Scanner(System.in);

      System.out.println("Entre com o lado A, B e C:");
      Double sides[] = {s.nextDouble(), s.nextDouble(), s.nextDouble()};
      Arrays.sort(sides, Collections.reverseOrder());

      System.out.print("Entrada: ");
      for(double side: sides) {
         System.out.print(side+ " ");
      }
      System.out.println();

      checkTriangulo(sides);

   }

   private static void checkTriangulo(Double[] sides){

      double ladoA = sides[0];
      double ladoB = sides[1];
      double ladoC = sides[2];

      if(ladoA >= (ladoB+ladoC)){
         System.out.println("NAO FORMA TRIANGULO");
      }else{

         if(pow(ladoA, 2) == pow(ladoB, 2) + pow(ladoC, 2)){
            System.out.println("TRIANGULO RETANGULO");
         }

         if(pow(ladoA, 2) > pow(ladoB, 2) + pow(ladoC, 2)){
            System.out.println("TRIANGULO OBTUSANGULO");
         }

         if(pow(ladoA, 2) < pow(ladoB, 2) + pow(ladoC, 2)){
            System.out.println("TRIANGULO ACUTANGULO");
         }

         if(ladoA == ladoB && ladoB == ladoC){
            System.out.println("TRIANGULO EQUILATERO");
         }else if(ladoA == ladoB || ladoB == ladoC || ladoC == ladoA){
            System.out.println("TRIANGULO ISOSCELES");
         }

      }

   }
}

File: challenges/3-tipos-de-triangulo/javascript/heltonp21/tiposDeTriangulo.js
let a = Number(process.argv.slice(2,3)); 
let b = Number(process.argv.slice(3,4));
let c = Number(process.argv.slice(4,5));

    if (a >= b + c || b >= a + c || c >= a + b) {
        console.log("NAO FORMA TRIANGULO.")
    }
    else {
        if (a**2 == b**2 + c**2 || b**2 == a**2 + c**2 || c**2 == a**2 + b**2) {
            console.log("TRIANGULO RETANGULO.")
        }
        else if (a**2 > b**2 + c**2 || b**2 > a**2 + c**2 || c**2 > a**2 + b**2) {
            console.log("TRIANGULO OBTUSANGULO.")
        }
        else if (a**2 < b**2 + c**2 || b**2 < a**2 + c**2 || c**2 < a**2 + b**2) {
            console.log("TRIANGULO ACUTANGULO.")
        }
      
        if (a === b && a === c) {
            console.log("TRIANGULO EQUILATERO.")
        }
        else if (a === b || a === c || b === c) {
            console.log("TRIANGULO ISOSCELES.")
        }        
    }